src/rust/engine/process_execution/sandboxer/src/lib.rs

```diff
@@ -208,3 +211,87 @@ impl SandboxerGrpc for SandboxerGrpcImpl {
         ret
     }
 }
+
+// The client that sends requests to the sandboxer service.
+#[derive(Clone, Debug)]
+pub struct SandboxerClient {
+    grpc_client: SandboxerGrpcClient<Channel>,
+}
+
+impl SandboxerClient {
+    pub async fn connect(socket_path: &Path) -> Result<Self, String> {
+        let socket_path_buf = socket_path.to_path_buf();
+        let channel = Endpoint::try_from("http://[::]:50051")
+            .map_err(|e| e.to_string())?
+            .connect_with_connector(service_fn(move |_: Uri| {
+                let socket_path_buf = socket_path_buf.clone();
+                async {
+                    Ok::<_, std::io::Error>(TokioIo::new(
+                        UnixStream::connect(socket_path_buf).await?,
+                    ))
+                }
+            }))
+            .await
+            .map_err(|e| e.to_string())?;
+        Ok(SandboxerClient {
+            grpc_client: SandboxerGrpcClient::new(channel),
+        })
+    }
+
+    pub async fn connect_with_retries(socket_path: &Path) -> Result<Self, String> {
+        // TODO: POLLING_INTERVAL seems to be a good time quantum to use when waiting
+        //  for server startup (since we know the server sleeps that long on startup)
+        //  but this is somewhat arbitrary and could be tuned further.
+        let sleep_time = SandboxerService::POLLING_INTERVAL;
+        let mut slept = Duration::ZERO;
+        let mut maybe_client: Result<Self, String> = Err("Not connected yet".to_string());
+
+        for _ in 0..20 {
```

Comment:
In practice it seems to take about 250ms for a server to become ready for connection. I still need to debug this since it really shouldn't take that long. It should be the 50ms the server sleeps on startup, plus some tiny overhead. But it seems tonic adds substantial overhead for whatever reason.

Reply:
What about reducing the connect's poll interval?

Reply:
I've experimented with that. It's 250ms actual time on the server, not an artifact of probe resolution.